RavenDB-14223 Various rollup and retention fixes

karmeli87 opened the pull request.

ppekrol commented on src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesConfiguration.cs:
```diff
@@ -8,11 +9,16 @@ public class TimeSeriesConfiguration : IDynamicJson
         internal const char TimeSeriesRollupSeparator = '@';
         public Dictionary<string, TimeSeriesCollectionConfiguration> Collections { get; set; }
 
+        public TimeSpan PolicyCheckFrequency = TimeSpan.FromSeconds(60);
```
Change to property

ppekrol commented on src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:
```diff
@@ -1620,6 +1620,78 @@ public void AddTimeSeriesNameToMetadata(DocumentsOperationContext ctx, string do
 
+        private static string EnsureOriginalName(DocumentsOperationContext ctx, string docId, string tsName)
+        {
+            var parts = tsName.Split(TimeSeriesConfiguration.TimeSeriesRollupSeparator);
```
Can we have '@' in the name of TS?

karmeli87 replied:
https://issues.hibernatingrhinos.com/issue/RavenDB-14800

karmeli87 committed:
RavenDB-14223 add rollup and retention frequancy check

karmeli87 committed:
RavenDB-14223 various rollup and retention fixes

- Ensure policy name preserve the original name
- Add logging
- Prevent from the retention to trigger the rollup

ppekrol commented:
I've rebased

karmeli87 committed:
RavenDB-14223 Make sure stats and the actual data are in sync

ppekrol commented on src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs:
```diff
@@ -70,7 +70,10 @@ public TimeSeriesStats(Transaction tx)
             var tableName = collection.GetTableName(CollectionTableType.TimeSeriesStats); // TODO: cache the collection and pass Slice
-            TimeSeriesStatsSchema.Create(tx, tableName, 16);
+            
+            if (tx.IsWriteTransaction)
```
I do not like this. Better to have 2 separate methods. Can't we have 2 separate methods? GetTable and GetOrCreateTable?

ppekrol committed:
RavenDB-14223 fixing compilation after rebase
